bluepysnap/_plotting.py

```diff
@@ -0,0 +1,363 @@
+
+        From notebooks:
+        >>> from IPython.display import HTML
+        >>> from bluepysnap import Simulation
+        >>> report = Simulation("config.json").spikes["my_population"]
+        >>> anim, ax = report.firing_animation()
+        >>> HTML(anim.to_html5_video())
+    """
+    plt = _get_pyplot()
+    from matplotlib.animation import FuncAnimation
+
+    def _check_axis(axis):
+        """Verifies axes values."""
+        axes = {Node.X, Node.Y, Node.Z}
+        if axis not in axes:
+            raise BluepySnapError('{} is not a valid axis'.format(axis))
+
+    _check_axis(x_axis)
+    _check_axis(y_axis)
+
+    spike_report = filtered_report.spike_report
+    population_names = filtered_report.spike_report.population_names
+    report = filtered_report.report
+
+    data = pd.DataFrame(index=report.index, columns=[x_axis, y_axis], dtype=np.float32)
+
+    for population in population_names:
+        spikes = spike_report[population]
+        pop_mask = report["population"] == population
+
+        ids = report.loc[pop_mask, "ids"].to_numpy()
+        try:
+            values = spikes.nodes.get(properties=[x_axis, y_axis]).loc[ids].to_numpy()
+            data.loc[pop_mask, [x_axis, y_axis]] = values
+        except BluepySnapError:
+            continue
+
+    data = data[data.notna()]
+
+    if ax is None:
+        fig = plt.figure()
+        ax = plt.gca()
+        ax.set_title('time = {}ms'.format(np.min(data.index)))
+        x_limits = [data[x_axis].min(), data[x_axis].max()]
+        y_limits = [data[y_axis].min(), data[y_axis].max()]
+        ax.set_xlim(*x_limits)
+        ax.set_ylim(*y_limits)
+        ax.set_xlabel('{} $\mu$m'.format(x_axis))  # noqa
+        ax.set_ylabel('{} $\mu$m'.format(y_axis))  # noqa
+
+    else:
+        fig = ax.figure
+
+    dots = ax.plot([], [], '.k')
+
+    def update_animation(frame):
+        """Update the animation plots and axes."""
+        ax.set_title('time = ' + str(frame * dt) + ' ms')
+        mask = (data.index >= frame * dt) & (data.index <= (frame + 1) * dt)
+        positions = data.loc[mask, [x_axis, y_axis]].values
+        x = positions[:, 0]
+        y = positions[:, 1]
+        dots[0].set_data(x, y)
+        return dots
+
+    frames = list(range(int(data.index[0] / dt), int(data.index[-1] / dt)))
+    anim = FuncAnimation(fig, update_animation, frames=frames)
+    return anim, ax
+
+
+def frame_trace(filtered_report, plot_type='mean', ax=None):  # pragma: no cover
+    """Return potential plot displaying the voltage as a function of time from a frame report.
+
+    Args:
+        plot_type (str): string either `all` or `mean`. `all` will plot the first 15 traces from the
+        group. `mean` will plot the mean value of the node
+        ax: A plot axis object that will be updated
+
+    Returns:
+        matplotlib.Axis: axis containing the soma's traces.
+    """
+    # pylint: disable=too-many-locals
+
+    plt = _get_pyplot()
+    max_per_pop = 15
```

Comment:
Could be defined next to the place it is used for the first time.

Reply:
done